pkg/validate/security_context_linux.go

```diff
@@ -876,6 +879,11 @@ var _ = framework.KubeDescribe("Security Context", func() {
 			if !supportsUserNamespaces {
 				Skip("no runtime handler found which supports user namespaces")
 			}
+
+			pathIDMap := rootfsPath(resp.GetInfo())
+			if err := supportsIDMap(pathIDMap); err != nil {
```

Comment:
Should this run once `BeforeAll` user namespace tests?

Reply:
We can't add BeforeAll() for not ordered containers. And we don't need them to be ordered, it will slow it down for no reason.

I was reading the ginkgo manual, I think it makes sense to do the following:
1. Add a `When("Host idmap mount support is needed")` and move the tests that need it under that
2. Add a BeforeEach for that, that checks for idmap mount support
3. Add a `When("Host idmap mount support is not needed")` and move the tests that don't need it there (like multiple mappings and all, idmap mounts is not needed, validation is expected to fail).
4. Make the detection of the runtime handler support for userns inside a sync.Once, so we just query the runtime once and work with the answer later.

I'll push it in a few minutes

Let me know what you think :)

Reply:
Sounds good to me, thank you!

Reply:
Pushed this now

Reply:
If you see the commit diff with `git show -w`, it will be cleaner. It seems bigger, but it is only nesting some tests under the `When("Host idmap mount support is not needed")`

Reply:
I'll re-validate this on the containerd side, just in case. I'll let you know when it is ready for review

Reply:
While doing that, I found there is a test that needs more care, as containerd runs the tests inside a userns and the whole UID space was hardcoded and that test failed.

I'll double check tomorrow all is fine for that case before asking for another review.